network_simulator/statistics.py

```diff
@@ -4,6 +4,186 @@
 """
 
 
+from .host import *
+from .link import *
+from .flow import *
+
+
+class LinkStats(object):
+    """
+    Contains information of a link regarding to per-link buffer occupancy,
+    packet loss, and flow rate.
+
+    :ivar list buffer_occupancy: a list of (timestamp, buffer_occupancy)
+    tuples.
+    :ivar list packet_loss_times: a list of timestamps, each corresponding
+    to a packet loss.
+    :ivar list packet_transmit_times: a list of timestamps corresponding
```

Comment:
I just realized this needs to be a list of (timestamp, packet_size) tuples, where the packet_size is in bits. Otherwise, we cannot calculate the flow rate based on just the timestamps.